logos-codegen/src/mir.rs

```diff
@@ -111,23 +111,38 @@ impl TryFrom<Hir> for Mir {
 
                 Ok(Mir::Alternation(alternation))
             }
-            HirKind::Literal(literal) => Ok(Mir::Literal(literal)),
+            HirKind::Literal(literal) => {
+                let s = std::str::from_utf8(&*literal.0).unwrap();
```

Comment:
Note, `HirKind::Literal` is now multiple characters. I'm not sure if it's required to be able to parse as utf8 or not, this may need to use a different approach if it's allowed arbitrary bytes.